adal/src/main/java/com/microsoft/aad/adal/TokenCacheItem.java

```diff
@@ -82,6 +84,9 @@ public TokenCacheItem() {}
         this.mIsMultiResourceRefreshToken = tokenCacheItem.getIsMultiResourceRefreshToken();
         this.mTenantId = tokenCacheItem.getTenantId();
         this.mFamilyClientId = tokenCacheItem.getFamilyClientId();
+        if (tokenCacheItem.isExtendedLifetimeValid()) {
```

Comment:
I think we should not assign the extendedExpiresOn to a new tokenCacheItem when we know it is no longer valid. What do you think?